### ISSUE TITLE
Fast Random Lookups in ETS

### ISSUE THREAD

**Issue by wfvining** [wfvining/Meridian#5]
Need a faster way to lookup a random genome from an ets table. The current implementation is O(n^d) where n is the granularity of the map and d is the dimensionality of the behavior space. Obviously this is totally unacceptable.

I'm not sure how to do this, but it will likely require and additional data structure of some sort that tracks the grid cells that are occupied. Updates can be broadcast from the master to all the workers (since the master receives an update from each worker when it has evaluated a new genome). Using these updates, a map can be made as follows:

``` erlang
N = ets:info(Map, size).
Index = #{ 1 => BehaviorGrid1, 2 => BehaviorGrid2, ... , N => BehaviorGridN }
I = rand:uniform(N),
Grid = map:get(I, Index).
```

This isn't quite sufficient, though. We need an inverted index, mapping grid to index as updates come in from other workers.

``` erlang
I = map:get(Grid, InvertedIndex).
map:put(I, Grid, Index).
```

or something like that.

**Comment by wfvining** [comment]
The approach above is problematic for two reasons:
1. The Grid cells in the map will almost never be the same as the grid cells in the table.
2. Must add extra code to ensure that all workers wait until the map is non empty, hence the workers have to be synchronize.
   
   better option might be to store the map in another ets table, or _as_ another ets table.

**Comment by wfvining** [comment]
Fixed in e544692 by adding two additional tables, an index mapping all integers 1..MapSize to a grid cell and an inverted index mapping grid cells to integers in the same range.
